beacon-chain/rpc/validator/exit.go

```diff
@@ -0,0 +1,75 @@
+package validator
+
+import (
+	"context"
+
+	ptypes "github.com/gogo/protobuf/types"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/go-ssz"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/statefeed"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/mathutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/roughtime"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// RequestExit requests an exit for a validator.
+func (vs *Server) RequestExit(ctx context.Context, req *ethpb.VoluntaryExit) (*ptypes.Empty, error) {
+	s, err := vs.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
+	}
+
+	// Confirm the validator is eligible to exit with the parameters provided
+	if req.ValidatorIndex >= uint64(len(s.Validators)) {
+		return nil, status.Errorf(codes.InvalidArgument, "Unknown validator index %d", req.ValidatorIndex)
+	}
+	validator := s.Validators[req.ValidatorIndex]
+	if !helpers.IsActiveValidator(validator, req.Epoch) {
+		return nil, status.Errorf(codes.InvalidArgument, "Validator %d not active at epoch %d", req.ValidatorIndex, req.Epoch)
+	}
+	if validator.ExitEpoch != params.BeaconConfig().FarFutureEpoch {
+		return nil, status.Errorf(codes.InvalidArgument, "Validator %d already exiting", req.ValidatorIndex)
+	}
+
+	secondsPerEpoch := params.BeaconConfig().SecondsPerSlot * params.BeaconConfig().SlotsPerEpoch
```

Comment:
This is fairly complex logic I'd advise placing in its own function, making it easier to write specific, table-driven tests for its behavior

Reply:
I'm thinking that a chunk of this code will be shared between this and the module that receives voluntary exits from the network and punted on where that code should live, but yes it makes sense to break this out now for clarity.

Reply:
Makes sense to refactor this to `/core/helper/exit`

Reply:
Willdo.

Reply:
This has been completed in latest commit.